pyne/alara.py

```diff
@@ -887,59 +1063,231 @@ def _get_subvoxel_array(mesh, cell_mats):
+        for dt in decay_times:
             # Skip "shutdown" string in list.
-            if dc == u'shutdown':
+            if str_to_unicode(dt) == u'shutdown':
                 continue
             # Convert to [s].
-            dc_s = _convert_unit_to_s(dc)
-            if idc_s == dc_s:
-                # idc_s matches dc_s. return original string, dc.
-                return dc
-            elif dc_s != 0.0 and (abs(idc_s - dc_s)/dc_s) < 1e-6:
-                return dc
-        # if idc doesn't match any string in phtn_src_dc list, raise an error.
+            dt_s = _convert_unit_to_s(dt)
+            if idt_s == dt_s:
+                # idt_s matches dt_s. return original string, dt.
+                return dt
+            elif dt_s != 0.0 and (abs(idt_s - dt_s)/dt_s) < 1e-6:
+                return dt
+        # if idt doesn't match any string in decay_times list, raise an error.
         raise ValueError(
-            'Decay time {0} not found in phtn_src file'.format(idc))
+            'Decay time {0} not found in decay_times'.format(idt))
+
+
+def responses_output_zone(responses=None, wdr_file=None, alara_params=None):
+    """
+    This function returns a string that is an output block for an alara input
+    file, configured for zone resolution.
+
+    Parameters
+    ----------
+    responses : list of string
+        A keyword represent the alara output zone. The following responses are
+        supported:
+            - decay_heat
+            - specific_activity
+            - alpha_heat
+            - beta_heat
+            - gamma_heat
+            - wdr
+            - photon_source
+    wdr_file : string
+        File name of the standard used to calculate wdr.
+    alara_params: string
+        Alara parameters
+
+    Returns
+    -------
+    String represent the output zone block corresponding to the response.
+    """
+
+    # set default value for functions do not need response
+    if responses == None:
+        return ''
+    # input check
+    for response in responses:
+        if response not in response_strings.keys():
+            raise ValueError('response {0} not supported.'.format(response))
+
+    output_strings = {"decay_heat": "      total_heat\n",
+                      "specific_activity": "      specific_activity\n",
+                      "alpha_heat": "      appha_heat\n",
+                      "beta_heat": "      beta_heat\n",
+                      "gamma_heat": "      gamma_heat\n"}
+    if 'wdr' in responses:
+        output_strings['wdr'] = ''.join(["       wdr ", wdr_file, "\n"])
+    if 'photon_source' in responses:
+        alara_lib = get_alara_lib(alara_params)
+        output_strings["photon_source"] =  ''.join(["      photon_source ",
+            alara_lib, " phtn_src 1 2e7\n"])
+    output_zone = ["output zone\n"]
+    for response in responses:
+        output_zone.append(output_strings[response])
+    output_zone.append("end")
+    return ''.join(output_zone)
```

Comment:
Maybe remove the "\n" from each string and use `"\n".join()`

Reply:
Thanks, I applied it in https://github.com/pyne/pyne/pull/1123/commits/bc19b1d3625988e768697adf61d4b7bf2ab19da4.